src/scripts/main.js

```diff
@@ -1,3 +1,254 @@
+        const shifted = mergeTiles(gameTable[i].filter((val) => val > 0));
+
+        if (shifted) {
+          for (let j = 0; j < shifted.length; j++) {
+            gameData[i][j] = shifted[j];
+          }
+        }
+      }
+      break;
+    case keyCode.right:
+      for (let i = 0; i < size; i++) {
+        const shifted = mergeTiles(gameTable[i].filter((val) =>
+          val > 0).reverse());
+
+        if (shifted) {
+          for (let j = 0; j < shifted.length; j++) {
+            gameData[i][size - 1 - j] = shifted[j];
+          }
+        }
+      }
+      break;
+    case keyCode.up:
+      for (let j = 0; j < size; j++) {
+        const arr = [];
+
+        for (let i = 0; i < size; i++) {
+          arr.push(gameTable[i][j]);
+        }
+
+        const mergedValues = mergeTiles(arr.filter((val) => val > 0));
+
+        if (mergedValues) {
+          for (let i = 0; i < mergedValues.length; i++) {
+            gameData[i][j] = mergedValues[i];
+          }
+        }
+      }
+      break;
+    case keyCode.down:
+      for (let j = 0; j < size; j++) {
+        const arr = [];
+
+        for (let i = 0; i < size; i++) {
+          arr.push(gameTable[size - 1 - i][j]);
+        }
+
+        const mergedValues = mergeTiles(arr.filter((val) => val > 0));
+
+        if (mergedValues) {
+          for (let i = 0; i < mergedValues.length; i++) {
+            gameData[size - 1 - i][j] = mergedValues[i];
+          }
+        }
+      }
+      break;
+    default:
+      break;
+  }
+
+  return gameData;
+}
+
+function gameEnd() {
+  if (`${gameTable}` !== `${updateGameBoard(keyCode.left)}`
+    || `${gameTable}` !== `${updateGameBoard(keyCode.right)}`
+    || `${gameTable}` !== `${updateGameBoard(keyCode.up)}`
+    || `${gameTable}` !== `${updateGameBoard(keyCode.down)}`) {
+    return true;
+  }
+}
+
+function updateScore() {
+  if (scoreNumber.length > 0) {
+    score += scoreNumber.reduce((acc, cur) => acc + cur);
+    scoreBoard.textContent = !score ? 0 : score;
+  }
+}
+
+function performAction(direct) {
+  scoreNumber = [];
+
+  const gameDataNext = updateGameBoard(direct);
+
+  if (`${gameTable}` === `${gameDataNext}`) {
+    return false;
+  }
+
+  gameTable = gameDataNext;
+
+  return true;
+}
+
+function displayGameResult(result) {
+  if (result) {
+    messageWin.classList.remove('hidden');
+  } else {
+    messageLose.classList.remove('hidden');
+  }
+  document.removeEventListener('keydown', handleKeyPress);
+}
+
+function startNewGame() {
+  button.classList.remove('start');
+  button.textContent = 'Restart';
+  button.classList.add('restart');
+  messageStart.classList.add('hidden');
+  messageWin.classList.add('hidden');
+  messageLose.classList.add('hidden');
+  scoreBoard.textContent = '0';
+  score = 0;
+  cleanTable();
+  randomNumber();
+  randomNumber();
+  updateGameDataUnpacked();
+  document.addEventListener('keydown', handleKeyPress);
+}
+
+function cleanTable() {
+  [...tableCells].map(el => {
```

Comment:
Use the `forEach` method instead of the `map` method, as you are not returning a new array.